File: src/com/prey/net/PreyHttpResponse.java
/*******************************************************************************
 * Created by Carlos Yaconi
 * Copyright 2012 Fork Ltd. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.net;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

import org.apache.http.HttpResponse;
import org.apache.http.StatusLine;

import com.prey.PreyLogger;

public class PreyHttpResponse {

	private StatusLine statusLine;
	private String responseAsString;
	private HttpResponse response;

	// public PreyHttpResponse(HttpMethod method){
	// this.statusLine = method.getStatusLine();
	//
	// try {
	// this.responseAsString =
	// convertStreamToString(method.getResponseBodyAsStream());
	// } catch (IOException e) {
	// Log.d(
	// "Can't receive body stream from http connection, setting response string as ''");
	// this.responseAsString = "";
	// }
	// }

	public PreyHttpResponse(HttpResponse response) {
		this.response=response;
		this.statusLine = response.getStatusLine();

		try {
			this.responseAsString = convertStreamToString(response.getEntity().getContent());
		} catch (IOException e) {
			PreyLogger.d("Can't receive body stream from http connection, setting response string as ''");
			this.responseAsString = "";
		}
	}

	public PreyHttpResponse(InputStream responseStream, StatusLine statusLine) {
		this.statusLine = statusLine;
		this.responseAsString = convertStreamToString(responseStream);
	}

	private String convertStreamToString(InputStream is) {
		BufferedReader reader = new BufferedReader(new InputStreamReader(is));
		StringBuilder sb = new StringBuilder();
		String line = null;
		try {
			while ((line = reader.readLine()) != null) {
				sb.append(line + "\n");
			}
		} catch (IOException e) {
			e.printStackTrace();
		} finally {
			try {
				is.close();
			} catch (IOException e) {
				e.printStackTrace();
			}
		}
		return sb.toString();
	}

	public StatusLine getStatusLine() {
		return statusLine;
	}

	public String getResponseAsString() {
		return responseAsString;
	}

	@Override
	public String toString() {

		return statusLine.toString() + " " + responseAsString;
	}

	public HttpResponse getResponse() {
		return response;
	}

	public void setResponse(HttpResponse response) {
		this.response = response;
	}

}

File: app/src/main/java/com/prey/net/kotlin/PreyHttpResponse.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.net.kotlin

import com.prey.kotlin.PreyLogger
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection


class PreyHttpResponse {
    private var statusCode = 0
    private var responseAsString: String? = null
    private var response: HttpURLConnection? = null
    private var mapHeaderFields: Map<String, List<String>>? = null

    constructor(connection: HttpURLConnection) {
        try {
            this.response = connection
            this.statusCode = connection.responseCode
            var input: InputStream? = null
            input =
                if (statusCode == HttpURLConnection.HTTP_OK || statusCode == HttpURLConnection.HTTP_CREATED) {
                    connection.inputStream
                } else {
                    connection.errorStream
                }
            this.responseAsString = convertStreamToString(input)
            PreyLogger.d("responseAsString:$responseAsString")
        } catch (e: IOException) {
            PreyLogger.d("Can't receive body stream from http connection, setting response string as ''")
            this.responseAsString = ""
        }
    }

    constructor(statusCode: Int, responseAsString: String) {
        try {
            this.response = null
            this.statusCode = statusCode
            this.responseAsString = responseAsString
            PreyLogger.d("responseAsString:$responseAsString")
        } catch (e: Exception) {
            PreyLogger.d("Can't receive body stream from http connection, setting response string as ''")
            this.responseAsString = ""
        }
    }

    constructor(
        statusCode: Int,
        responseAsString: String,
        mapHeaderFields: Map<String, List<String>>?
    ) {
        try {
            this.response = null
            this.statusCode = statusCode
            this.responseAsString = responseAsString
            this.mapHeaderFields = mapHeaderFields
            PreyLogger.d("responseAsString:$responseAsString")
        } catch (e: Exception) {
            PreyLogger.d("Can't receive body stream from http connection, setting response string as ''")
            this.responseAsString = ""
        }
    }

    private fun convertStreamToString(`is`: InputStream?): String? {
        var out: String? = null
        try {
            val rd = BufferedReader(InputStreamReader(`is`))
            var line: String?
            val response = StringBuffer()
            while ((rd.readLine().also { line = it }) != null) {
                response.append(line)
                response.append('\r')
            }
            rd.close()
            out = response.toString()
        } catch (e: Exception) {
            PreyLogger.e("Error:" + e.message, e)
        }
        return out
    }

    fun getStatusCode(): Int {
        return statusCode
    }

    fun getResponseAsString(): String? {
        return responseAsString
    }

    override fun toString(): String {
        return "$statusCode $responseAsString"
    }

    fun getResponse(): HttpURLConnection? {
        return response
    }

    fun setResponse(response: HttpURLConnection?) {
        this.response = response
    }

    fun getMapHeaderFields(): Map<String, List<String>>? {
        return mapHeaderFields
    }
}